lib/adagrams.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+# Wave 1
+def draw_letters
+  letters_pool_to_draw_from = {
+    'A' => 9,
+    'B' => 2,
+    'C' => 2,
+    'D' => 4,
+    'E' => 12,
+    'F' => 2,
+    'G' => 3,
+    'H' => 2,
+    'I' => 9,
+    'J' => 1,
+    'K' => 1,
+    'L' => 4,
+    'M' => 2,
+    'N' => 6,
+    'O' => 8,
+    'P' => 2,
+    'Q' => 1,
+    'R' => 6,
+    'S' => 4,
+    'T' => 6,
+    'U' => 4,
+    'V' => 2,
+    'W' => 2,
+    'X' => 1,
+    'Y' => 2,
+    'Z' => 1
+  }
+
+  ten_strings_array = []
+
+  letters_pool_keys = letters_pool_to_draw_from.keys
+  until ten_strings_array.length == 10
+    rand_letter = letters_pool_keys[rand(letters_pool_keys.size)]
+    if (letters_pool_to_draw_from[rand_letter]).zero?
+      next
+    else
```

Comment:
This runs, but it gives an equal probability of drawing a `Z` and an `E`.  

A better strategy would be to use your hash to build an array of all the letters with 9 As, 2 Bs etc , shuffle the array and then draw the right number of letters.